subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/DefaultTransformerInvoker.java

```diff
@@ -203,25 +212,60 @@ public int hashCode() {
         private final File outputDir;
         private final File resultsFile;
         private final HashCode persistentCacheKey;
+        private final ComponentArtifactIdentifier artifactId;
+        private final ResolvableDependencies incoming;
 
-        public TransformerExecution(File primaryInput, Transformer transformer, HashCode persistentCacheKey, File workspace) {
+        public TransformerExecution(File primaryInput, Transformer transformer, HashCode persistentCacheKey, File workspace, @Nullable ComponentArtifactIdentifier artifactId, ResolvableDependencies incoming) {
```

Comment:
We don't use `incoming` for anything if `artifactId` is `null`, so we could as well skip both. I think it would be better to pass something that is basically a `Supplier<FileCollection>` -- for the no `artifactId` case it could return an empty collection always.